doc/source/compression.rst

```diff
@@ -0,0 +1,154 @@
+Compression
+===========
+
+In this tutorial you will learn all about compression in TileDB. It is
+highly recommended that you read the tutorials on attributes and
+tiling dense/sparse arrays first.
+
+
+Basic concepts and definitions
+------------------------------
+
+.. toggle-header::
+    :header: **Attribute-based compression**
+
+    TileDB allows you to choose a different compression scheme for
+    each attribute.
+
+.. toggle-header::
+    :header: **Tile-based compression**
+
+    The data tile is the atomic unit of compression, i.e., each
+    data tile is compressed separately.
+
+.. toggle-header::
+    :header: **Compression tradeoff**
+
+    There is no ideal compressor. Each compressor offers a tradeoff
+    between compression ratio and compression/decompression speed,
+    which highly depends on the array data of your application.
+
+Compressing arrays
+------------------
+
+The data types and values of the array cells most likely vary across attributes.
+Therefore, it is desirable to allow defining a different compression scheme
+per attribute, since different compressors may be more effective for different
+types of data. Here is how you specify the compressor when defining an attribute
+upon array creation:
+
+.. content-tabs::
+
+   .. tab-container:: cpp
+      :title: C++
+
+      .. code-block:: c++
+
+        Context ctx;
+        ArraySchema schema(ctx, TILEDB_DENSE);
+        auto a = Attribute::create<int>(ctx, "a");
+        a.set_compressor({TILEDB_GZIP, -1});
+        schema.add_attribute(a);
+
+TileDB offers a variety of compressors to choose from (see below). Moreover,
+you can adjust the compression level (the second argument in the function
+setter above - ``-1`` means "default compression level"). If a compressor
+is not specified, TileDB will use no compression by default.
+
+When compressing the data tiles of an attribute, TileDB stores some
+necessary metadata in file ``__fragment_metadata.tdb``, such as the
+starting location of each compressed tile and the original tile size
+in the case of variable-length attributes (recall that the original tile
+size has fixed size for fixed-length attributes in *both* dense and
+sparse arrays).
+
+You can also specify a different compressor for the coordinates
+tiles as follows (the default is ZSTD with default compression level):
+
+.. content-tabs::
+
+   .. tab-container:: cpp
+      :title: C++
+
+      .. code-block:: c++
+
+        schema.set_coords_compressor({TILEDB_LZ4, -1});
+
+To maximize the effect of compression on coordinates, TileDB first
+unzips the coordinates tuples and groups the coordinates along
+the dimensions. For instance, coordinates ``(1,2)``, ``(1,3)``, ``(1,5)``
+will be laid out as ``1 1 1 2 3 5`` prior to passing them into
+the compressor. Placing "similar" values next to each generally leads
```

Comment:
I would say, sorting allows for slowly varying coordinate values to be grouped together and compressed.